### ISSUE TITLE
Support AWS eu-north-1

### ISSUE THREAD

**Issue by bucht** [kubernetes/kops#6204]
Add support for the new AWS region eu-north-1 with 3 AZ:
https://aws.amazon.com/blogs/aws/now-open-aws-europe-stockholm-region/

**Comment by willnewby** [comment]
I'd like to try and work on this, since it seems like a good first-time issue. 

@bucht if I run kops create like so:

```bash
ZONES="eu-north-1a,eu-north-1b,eu-north-1c"
kops create cluster \
     --cloud aws \
     --zones $ZONES \
     --master-zones $ZONES
```

Then creating the cluster works for me. However, the `--cloud aws` flag is required, since kops doesn't currently support inferring AWS from the eu-north-1 zones.

**Comment by bucht** [comment]
https://github.com/kubernetes/kops/blob/master/upup/pkg/fi/cloud.go
Seems to be the magic file :)

**Comment by willnewby** [comment]
@bucht If you specify the create command the way I did above, subscribe to Debian 9 in the AWS marketplace (https://aws.amazon.com/marketplace/server/procurement?productId=572488bb-fc09-4638-8628-e1e1d26436f4 ) and then use ami-e133bc9f (which is the AMI ID for Debian 9 in eu-north-1), your cluster should work. I'd say use this image until there's an official image posted for KOPS in eu-north-1.

I'm also working on a PR to default the cloud provider with the new AZs.

**Comment by nikoul** [comment]
@willnewby Using your tip with the Debian 9 image does not work.
It got deployed, but in the logs of all nodes I have : 

``` Dec 19 09:08:51 ip-1xx-xx-xx-xx nodeup[602]: W1219 09:08:51.174197     602 main.go:142] got error running nodeup (will retry in 30s): error loading Cluster "s3://KOPS-BUCKET/cluster.spec": eu-north-1 is not a valid region ```

**Comment by willnewby** [comment]
@nikoul thanks for the heads up, I'll investigate further.

**Comment by oekarlsson** [comment]
I tried this on the new kops 1.11, but no luck. I could get pass the "image not found" error using the stock debian image, but when the instances comes up it fails to run nodeup with "not a valid region". Can we expect this to work in 1.12 or will there be a workaround that we pehaps can try sooner?

**Comment by pliljenberg** [comment]
@svanlund I built `nodeup` (from the master branch) and like you got passed the `nodeup` step.
Now stuck at:
`error reading addons from "s3://example-bucket/test-sthlm-cluster/addons/bootstrap-channel.yaml": eu-north-1 is not a valid region`

This comes from running `protokube` docker contaner which runs the `channels` command internally. Building `channels` on the master branch and inserting that in the container seems to work and my cluster is up and running..
Not sure if there is a working version of `protokube` (1.11.0 was used in my current setup) that works or not. Will continue to investigate.

**Comment by yazzzir** [comment]
I'm running kops 1.11 in eu-north-1.
I have tested a coreos ami as well as a debian ami and both have the same behavior.

Everything seems to work OK, however the nodeup doesn't seem to run automatically or hangs? For a 3 master 5 node deployment (gossip DNS), the api ELB reports 0 of 3 healthy masters for about 10min after cluster creation (I only waited 10min). However when I create a bastion ig, jump into the masters and manually run nodeup on the masters - the ELB starts to report healthy instance.

At which point a kubectl get nodes only returns the master. Therefore I suspect a manual run of nodeup will also bring the workers online.

Curious whether I'm missing something. Previously my experience in eu-west-1 was end to end without intervention.
